tests/functional/helpers.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import time
+
+import pytest
+
+import gitlab.base
+
+SLEEP_INTERVAL = 0.5
+TIMEOUT = 60  # seconds before timeout will occur
+
+
+def delete_object(
+    *, object: gitlab.base.RESTObject, index: int, hard_delete: bool = False
+):
+    if index:
+        logging.info(f"Attempt {index+1} to delete {object!r}.")
+    try:
+        if hard_delete:
+            object.delete(hard_delete=True)
+        else:
+            object.delete()
+    except gitlab.exceptions.GitlabDeleteError:
+        logging.info(f"{object!r} already deleted.")
+        pass
+
+
+def safe_delete(
+    *,
+    manager: gitlab.base.RESTManager,
+    object_id: int,
+    description: str,
```

Comment:
I just realized we basically don't need to pass the manager, id and description actually. Each object has a `self.manager`, `self.id` and `self.__class__.__name__` so we could potentially just pass the object to this and handle it inside the delete function - could be less error prone.